### ISSUE TITLE
getUsersOfRoom broken in 0.73.0

### ISSUE THREAD

**Issue by wreiske** [RocketChat/Rocket.Chat#13041]
### Description:

After updating to 0.73.0, I can no longer see the members list in a group

### Steps to reproduce:

1. Upgrade to 0.73.0
2. Click on member's list in a group

### Expected behavior:

Show the group member list.

## CLIENT
```
Exception in delivering result of invoking 'getUsersOfRoom': TypeError: Cannot read property 'records' of undefined
    at o.call (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:467)
    at n._callback (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:1)
    at n._maybeInvokeCallback (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:61)
    at n.receiveResult (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:61)
    at w._livedata_result (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:61)
    at w.onMessage (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:61)
    at currentStatus.connected.forEachCallback.e (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:59)
    at Array.forEach (<anonymous>)
    at l.forEachCallback (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:59)
    at k.socket.onmessage.t [as onmessage] (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:59)
```

![screen shot 2018-12-27 at 10 12 24 pm](https://user-images.githubusercontent.com/6295044/50501526-d16b0e80-0a26-11e9-8a22-f562df79be1c.png)
![screen shot 2018-12-27 at 10 15 02 pm](https://user-images.githubusercontent.com/6295044/50501527-d16b0e80-0a26-11e9-9f48-10a7a347b4d7.png)

```webview: Exception in delivering result of invoking 'getUsersOfRoom':```

## SERVER
```
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: Exception while invoking method 'getUsersOfRoom' { MongoError: Unrecognized pipeline stage name: '$replaceRoot'
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: at queryCallback (/opt/Rocket.Chat/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb-core/lib/cursor.js:248:25)
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: at /opt/Rocket.Chat/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb-core/lib/connection/pool.js:532:18
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: at _combinedTickCallback (internal/process/next_tick.js:131:7)
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: at process._tickDomainCallback (internal/process/next_tick.js:218:9)
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: => awaited here:
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: at Function.Promise.await (/opt/Rocket.Chat/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:56:12)
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: at Promise.asyncApply (server/methods/getUsersOfRoom.js:23:11)
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: at /opt/Rocket.Chat/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: ok: 0,
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: errmsg: 'Unrecognized pipeline stage name: \'$replaceRoot\'',
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: code: 16436,
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: name: 'MongoError',
Dec 27 23:28:56 chat.nope.com rocketchat[18403]: [Symbol(mongoErrorContextSymbol)]: {} }
```
### Server Setup Information:

- Version of Rocket.Chat Server: 0.73.0 8442666b51c7daff054a841a5e3c757f89b29bbc
- Operating System: CentOS 7
- Deployment Method: tar
- Number of Running Instances: 4
- DB Replicaset Oplog: Yes
- NodeJS Version: v8.9.3
- MongoDB Version: v3.2.21 

Possibly related to https://github.com/RocketChat/Rocket.Chat/pull/12566

**Comment by geekgonecrazy** [comment]
@rodrigok @sampaiodiego 🔝

**Comment by geekgonecrazy** [comment]
Posting here for completeness.  Updating to mongo 3.4 will resolve.

https://docs.mongodb.com/manual/reference/operator/aggregation/replaceRoot/ Docs with note of required version.  Will need to update docs or bug fix to remove

**Comment by TomaszDom** [comment]
Guys, first replicaset number prefix breaks installations with no warning, now a core dependency is changed without a warning? And why does it matter that the mongodb release reached EOL, but you still require an unpatched nodejs version from September 2018. Again, not cool!

**Comment by sampaiodiego** [comment]
sry @TomaszDom you're right. we didn't take enough care about this one, but as you noticed, we released 0.73.0 last night and are already working to make it land fine to as many users as we can..

the other issues you mentioned are out of our control though:

* meteor is not updating the nodejs version as you can [see here](https://github.com/meteor/meteor/blob/devel/History.md#changes-2)
> Although Node 8.12.0 has been released, Meteor 1.8 still uses Node 8.11.4, due to concerns about excessive garbage collection and CPU usage in production.
* the replSet issue was introduced by mongo driver, which was also updated by meteor, and you can see the amazing investigation done by @geekgonecrazy on this issue https://github.com/meteor/meteor/issues/10362

**Comment by TwizzyDizzy** [comment]
Hi everybody,

I cannot confirm, that the fix is working in `0.74.0-rc.0`. I'm running MongoDB 2.6, though, not 3.2 as described here.

Log output on clicking on the members icon:

```
Jan 22 14:38:54 dev-chat RocketChat: Exception while invoking method 'getUsersOfRoom' { MongoError: exception: Unrecognized pipeline stage name: '$lookup'
Jan 22 14:38:54 dev-chat RocketChat: at queryCallback (/opt/rocketchat/bundle/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb-core/lib/cursor.js:248:25)
Jan 22 14:38:54 dev-chat RocketChat: at /opt/rocketchat/bundle/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb-core/lib/connection/pool.js:532:18
Jan 22 14:38:54 dev-chat RocketChat: at _combinedTickCallback (internal/process/next_tick.js:132:7)
Jan 22 14:38:54 dev-chat RocketChat: at process._tickDomainCallback (internal/process/next_tick.js:219:9)
Jan 22 14:38:54 dev-chat RocketChat: => awaited here:
Jan 22 14:38:54 dev-chat RocketChat: at Function.Promise.await (/opt/rocketchat/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:56:12)
Jan 22 14:38:54 dev-chat RocketChat: at Promise.asyncApply (server/methods/getUsersOfRoom.js:23:11)
Jan 22 14:38:54 dev-chat RocketChat: at /opt/rocketchat/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40
Jan 22 14:38:54 dev-chat RocketChat: errmsg: 'exception: Unrecognized pipeline stage name: \'$lookup\'',
Jan 22 14:38:54 dev-chat RocketChat: code: 16436,
Jan 22 14:38:54 dev-chat RocketChat: ok: 0,
Jan 22 14:38:54 dev-chat RocketChat: name: 'MongoError',
Jan 22 14:38:54 dev-chat RocketChat: [Symbol(mongoErrorContextSymbol)]: {} }
```

Please let me know, whether I should open a new issue for this one or whether we'll handle it here :)

Greetings to Porto Alegre!
Thomas

**Comment by TwizzyDizzy** [comment]
Ha! My bad. Then, indeed, I'll have to schedule a migration from 2.6 to >= 3.2. Closing again.

Cheers
Thomas
